homeassistant/components/olarm_sensors/config_flow.py

```diff
@@ -0,0 +1,271 @@
+            if json is None:
+                errors[AUTHENTICATION_ERROR] = "Invalid credentials!"
+
+            # If there are errors, show the setup form with error messages
+            if errors:
+                return await self._show_setup_form(errors=errors)
+
+            setup_devices = [dev["deviceName"] for dev in json]
+
+            # If there are no errors, create a config entry and return
+            firmware = json[0]["deviceFirmware"]
+            temp_entry = ConfigEntry(
+                domain=DOMAIN,
+                source="User",
+                version=1,
+                title="Olarm Sensors",
+                data={
+                    CONF_API_KEY: api_key,
+                    CONF_SCAN_INTERVAL: scan_interval,
+                    CONF_DEVICE_FIRMWARE: firmware,
+                    CONF_ALARM_CODE: alarm_code,
+                    CONF_OLARM_DEVICES: setup_devices,
+                    OLARM_DEVICES: json,
+                    OLARM_DEVICE_AMOUNT: len(json),
+                    OLARM_DEVICE_NAMES: setup_devices,
+                },
+            )
+
+            for device in json:
+                if device["deviceName"] not in setup_devices:
+                    continue
+
+                await asyncio.sleep(2)
+                coordinator = OlarmCoordinator(
+                    self.hass,
+                    entry=temp_entry,
+                    device_id=device["deviceId"],
+                    device_name=device["deviceName"],
+                    device_make=device["deviceAlarmType"],
+                )
+
+                await coordinator.update_data()
+
+                self.hass.data[DOMAIN][device["deviceId"]] = coordinator
+
+            # Saving the device
+            return self.async_create_entry(
+                title="Olarm Sensors",
+                data={
+                    CONF_API_KEY: api_key,
+                    CONF_SCAN_INTERVAL: scan_interval,
+                    CONF_DEVICE_FIRMWARE: firmware,
+                    CONF_ALARM_CODE: alarm_code,
+                    CONF_OLARM_DEVICES: setup_devices,
+                    OLARM_DEVICES: json,
+                    OLARM_DEVICE_AMOUNT: len(json),
+                    OLARM_DEVICE_NAMES: setup_devices,
+                },
+            )
+
+        return self.async_show_form(step_id="user", data_schema=self._get_schema())
+
+    def _get_schema(self):
+        """Return the data schema for the user form."""
+        return vol.Schema(
+            {
+                vol.Required(
+                    CONF_API_KEY,
+                    msg="The api key for your account.",
+                    description={
+                        "suggested_value": "Your Olarm API key",
+                        "description": "API key for accessing the Olarm API. You can find your API key here: https://user.olarm.co/#/api",
+                    },
+                ): cv.string,
+                vol.Required(
+                    CONF_SCAN_INTERVAL,
```

Comment:
We no longer accept the possibility of changing the scan interval via the config flow.
Please use a fixed update interval. If the user wants a different update interval, he/she can create an automation that calls `homeassistant.update_entity` on their specific needs.

See also https://developers.home-assistant.io/docs/integration_fetching_data/#polling-api-endpoints